src/Utils/utils.ts

```diff
@@ -544,3 +544,58 @@ export const fahrenheitToCelsius = (fahrenheit: number) => {
 export const keysOf = <T extends object>(obj: T) => {
   return Object.keys(obj) as (keyof T)[];
 };
+
+// Capitalizes the first letter of each word in a string, handling edge cases
+export const startCase = (str: string): string => {
+  if (!str || str.length === 0) return "";
+  return str
+    .replace(/([a-z])([A-Z])/g, "$1 $2")
+    .replace(/[_-]+/g, " ")
+    .replace(/\s+/g, " ")
+    .trim()
+    .split(" ")
+    .map((word) => word.charAt(0).toUpperCase() + word.slice(1).toLowerCase())
+    .join(" ");
+};
+
+// Converts a string to camelCase format, first word to lowercase and each subsequent word - uppercase letter, with no spaces.
+export const camelCase = (str: string): string => {
+  if (!str || str.length === 0) return "";
+  return str
+    .trim()
+    .toLowerCase()
+    .replace(/[-_\s]+(.)?/g, (_, c) => (c ? c.toUpperCase() : ""));
+};
+
+export function setNestedValueSafely(
```

Comment:
Let's replace all usages of setNestedValueSafely with proper type-safe change handlers. And not have this helper function altogether.